cdk/test/functions/budget.test.ts

```diff
@@ -0,0 +1,50 @@
+import * as path from 'path';
+import { App, Stack, StackProps } from 'aws-cdk-lib';
+import { Template } from 'aws-cdk-lib/assertions';
+import { CfnInclude } from 'aws-cdk-lib/cloudformation-include';
+import { BUNDLING_STACKS } from 'aws-cdk-lib/cx-api';
+import { Construct } from 'constructs';
+import { BudgetStack } from '../../src/stacks/budget';
+
+export class UnderTestStack extends Stack {
+  public readonly inner: Stack;
+  constructor(scope: Construct, id: string, props: StackProps) {
+    super(scope, id, props);
+    this.inner = new BudgetStack(this, 'stack', {});
+  }
+}
+
+export class OriginalStack extends Stack {
+  constructor(scope: Construct, id: string, props: StackProps) {
+    super(scope, id, props);
+    new CfnInclude(this, 'BudgetTemplate', {
+      templateFile: path.join(__dirname, '..', '..', '..', 'templates', 'budget.yaml'),
+    });
+  }
+}
+
+const context = {
+  [BUNDLING_STACKS]: [],
+};
+
+describe('resources', () => {
```

Comment:
I don't see where we actually compare the CFN resources with the ones synthezised from CDK. Such as https://github.com/superwerker/superwerker/blob/6e2af7291a22fe10617963d8321bc308d46c4461/cdk/test/guardduty.test.ts#L67